smithy-model/src/main/java/software/amazon/smithy/model/shapes/Shape.java

```diff
@@ -108,39 +109,69 @@ protected void validateMixins(Map<ShapeId, Shape> mixins, Map<ShapeId, Trait> in
         }
     }
 
-    protected MemberShape getRequiredMixinMember(
+    protected Map<String, MemberShape> getRequiredMembers(
```

Comment:
I think this could be a `MemberShape[]` since we don't really need a dynamically sized List, nor do we need a Map since the caller of this methods knows the order of the provided member names.